Library/Homebrew/livecheck.rb

```diff
@@ -87,20 +87,25 @@ def regex(pattern = nil)
   # that the formula/cask/resource should be skipped and the `skip_msg` very
   # briefly describes why it is skipped (e.g. "No longer developed or
   # maintained").
-  #
-  # @param skip_msg [String] string describing why the formula/cask is skipped
-  # @return [Boolean]
-  def skip(skip_msg = nil)
-    if skip_msg.is_a?(String)
-      @skip_msg = skip_msg
-    elsif skip_msg.present?
-      raise TypeError, "Livecheck#skip expects a String"
-    end
+  sig {
+    params(
+      # String describing why the formula/cask is skipped.
+      skip_msg: String,
+    ).returns(T::Boolean)
+  }
+  def skip(skip_msg = T.unsafe(nil))
+    @skip_msg = skip_msg if skip_msg.is_a?(String)
 
     @skip = true
   end
 
+  # A very brief description of why the formula/cask/resource is skipped (e.g.
+  # `No longer developed or maintained`).
+  sig { returns(T.nilable(String)) }
+  attr_reader :skip_msg
```

Comment:
I'd prefer we follow most style guides and leave this where it was:

https://rubystyle.guide/#consistent-classes
https://docs.rubocop.org/rubocop/cops_layout.html#layoutclassstructure

Reply:
I find it easier to read when `skip` (which also sets `@skip_msg`) is next to this.

Reply:
I tend to agree with @dduugg. I expect to see `attr_*` calls where this originally was and that's where I look for them. I understand your reasoning for this modified location but this trips me up.

Reply:
@reitermarkus Happy to merge this PR if this is put back, thanks.